### ISSUE TITLE
API: check if each onboarding task is complete

### ISSUE THREAD

**Issue by jaclync** [woocommerce/woocommerce-ios#8892]
Jorge will post about the endpoint to use. pe5sF9-175-p2#comment-1722

- [x] Networking & Yosemite layers https://github.com/woocommerce/woocommerce-ios/pull/9017

Done in https://github.com/woocommerce/woocommerce-ios/pull/9033
- [x] Integrate with the checklist
- [x] Refresh the task list whenever the view is displayed to the user

Done in https://github.com/woocommerce/woocommerce-ios/pull/9114
- [x] Refresh the task list when the user pulls to refresh

**Comment by jaclync** [comment]
Hey @selanthiraiyan I added a subtask `Refresh the task list on completion of certain tasks where the state might change` to the subtasks as another trigger to refresh the tasks. I started working on https://github.com/woocommerce/woocommerce-ios/issues/8908, and there's a completion block when a product is added and we can refresh the tasks again to reflect the potential state change. Feel free to move it to a separate issue or update anything!
